### PR TITLE
`Programming exercises`: Add dropdown for choosing between https and ssh in clone dialog

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Adds the dropdown as discussed in #4507 for the git protocol for programming exercises

### Description

The git protocol can now be chosen via a dropdown menu instead of a button. This makes accidental changes harder which should be more user friendly for new students, while not reducing the user experience for more experienced users.

Additionally fixes the position of the overlay: On develop the overlay for cloning is on the far left of the page instead of to the right of the "Clone Repository" button.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to the programming exercise
3. Try to clone the exercise
4. Try to change the git protocol and use ssh/https (whatever you had not chosen before)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Bildschirmfoto 2022-01-01 um 14 01 00
Bildschirmfoto 2022-01-03 um 22 08 55

### CODE REVIEW

**Comment by Strohgelaender** [comment]
Tested on TS1, the dropdown is shown and the buttons work. I have one suggestion:

Can you align the end of the buttons in the dropdown with the end of the dropdown itself? Currently, the buttons even exceed the size of the clone window. 
grafik

**Comment by krusche** [comment]
Thanks for working on this issue! Could you please design it closer to how Bitbucket is implementing the user interface?

image

So instead of "Git Protocol", use the one which is currently chosen. And in the dropdown, please simply use HTTP and SSH instead of "Use HTTP" and "Use SSH". Then we also don't need a translation.

We should also think about moving it to the left (as Bitbucket), which might be more natural

File: src/main/webapp/i18n/de/exercise-actions.json [hunk 1]
```diff
@@ -26,6 +26,9 @@
             "copiedUrl": "URL kopiert",
             "uploadFile": "Datei hochladen",
             "viewTeam": "Team",
+            "gitProtocol": "Git-Protokoll",
+            "useHTTPS": "HTTPS nutzen",
+            "useSSH": "SSH nutzen",
```
Comment (Strohgelaender) [inline src/main/webapp/i18n/de/exercise-actions.json:31]:
```suggestion
```

File: src/main/webapp/i18n/en/exercise-actions.json [hunk 2]
```diff
@@ -26,6 +26,9 @@
             "copiedUrl": "Copied URL",
             "uploadFile": "Upload a file",
             "viewTeam": "Team",
+            "gitProtocol": "Git protocol",
+            "useHTTPS": "Use HTTPS",
+            "useSSH": "Use SSH",
```
Comment (Strohgelaender) [inline src/main/webapp/i18n/en/exercise-actions.json:31]:
```suggestion
```

**Review by Strohgelaender — APPROVED** [review]
Tested on TS1, works as expected.

**Review by Strohgelaender — DISMISSED** [review]
Re-Tested on TS1, works 👍 Code changes look good to me.

**Review by baranimo7 — DISMISSED** [review]
tested on ts1, works as expected

**Review by MertAksehirlioglu — DISMISSED** [review]
Tested on ts2, works as expected

**Review by baranimo7 — APPROVED** [review]
Tested on TS1, works as expected

**Review by willergeorg — COMMENTED** [review]
Tested on TS1 and worked good. The fact that the size of the dropdown menu changes if SSH is selected is a bit weird, but I neither was able to identify the reason via a quick check nor to unify the alignment via the dev tools. At the end of the day, I do not think it is a crucial detail that needs immediate attention. Codewise I have one suggestion: Move the styling inside the html template into dedicated CSS classes or use bootstrap classes if possible. Besides that, everything else is fine. 👍

**Pull request merged**